src/Support/SupportData.php

```diff
@@ -153,14 +148,24 @@ public function send_data() {
 			sprintf( '%s/api/v1/support/', $endpoint ),
 			[
 				// We need long timeout here, in case the data being sent is large or the network connection is slow.
-				'timeout'  => 3000, // phpcs:ignore WordPressVIPMinimum.Performance.RemoteRequestTimeout.timeout_timeout
+				'timeout'  => 30, // phpcs:ignore WordPressVIPMinimum.Performance.RemoteRequestTimeout.timeout_timeout
 				'body'     => $data,
 				'compress' => true,
 			]
 		);
 
 		if ( ! is_wp_error( $response ) ) {
-			$response = json_decode( wp_remote_retrieve_body( $response ), true );
+
+			$response_body   = wp_remote_retrieve_body( $response );
+			$response        = json_decode( $response_body, true );
+			$json_last_error = json_last_error();
+
+			if ( JSON_ERROR_NONE !== $json_last_error ) {
+				return new WP_Error(
+					$json_last_error,
```

Comment:
`$json_last_error` would be an integer here, so it would not give any real meaning if this error were to be logged. I'd recommend using a unique error code to identify where the error came from.

```suggestion
					'malformed_json_amp_insight_endpoint,
```